packages/interactivity/src/hooks.tsx

```diff
@@ -139,8 +139,15 @@ const namespaceStack: string[] = [];
  *                  function exists is used.
  * @return The context content.
  */
-export const getContext = < T extends object >( namespace?: string ): T =>
-	getScope()?.context[ namespace || getNamespace() ];
+export const getContext = < T extends object >( namespace?: string ): T => {
+	const scope = getScope();
+	if ( ! scope ) {
+		throw Error(
+			'Cannot call `getContext()` outside getters and actions used by directives.'
+		);
+	}
```

Comment:
It’s not just getters and actions, so we should find a more generalistic way to say this and add possible solution instructions. Apart from that, can we show the error only in development? 

Something like this (I'm not sure about the correct syntax):

```suggestion
	if ( globalThis.SCRIPT_DEBUG && ! scope ) {
		throw Error(
			'Cannot call `getContext()` when there is no scope. If you are using an async function, please consider using a generator instead. If you are using some sort of async callbacks, like `setTimeout`, please wrap the callback with `withScope(callback)`.'
		);
	}
```

We should do the same in the `getElement` function.

To be honest, I'm not sure when we should throw errors in production and when we shouldn't. I think we haven't been very consistent in that regard. Perhaps we should do a general review of all these errors to have a more consistent policy.

Reply:
I'll include your suggestion. We can do the general review later. 🙂

Reply:
Thanks, I have included it in the [Tracking Issue](https://github.com/WordPress/gutenberg/issues/63232):  "Review all errors and make them more consistent"